src/strands/models/bedrock.py

```diff
@@ -112,7 +112,21 @@ def __init__(
         session = boto_session or boto3.Session(
             region_name=region_name or os.getenv("AWS_REGION") or "us-west-2",
         )
-        client_config = boto_client_config or BotocoreConfig(user_agent_extra="strands-agents")
+
+        # Add strands-agents to the request user agent
+        if boto_client_config:
+            existing_user_agent = getattr(boto_client_config, "user_agent_extra", None)
+
+            # Append 'strands-agents' to existing user_agent_extra or set it if not present
+            if existing_user_agent:
+                new_user_agent = f"{existing_user_agent} strands-agents"
```

Comment:
Any concern that there's effectively not a way to turn this off in this case?

Reply:
That's pretty common. For example, the SDKs themselves add a bunch to the user agents and you can't turn it off